src/objc/ModalWebViewController.mm

```diff
@@ -91,14 +93,38 @@ - (void)close {
   [self dismissViewControllerAnimated:YES completion:nil];
 }
 
+- (void)addUrl:(NSString *)urlToAdd {
+  if (self.visitedUrls.count == 0 ||
+      ![self.visitedUrls.lastObject isEqualToString:urlToAdd]) {
+    [self.visitedUrls addObject:urlToAdd];
+  }
+}
+
 #pragma mark - WKNavigationDelegate Methods
 
+// Called when the web view starts to load a page
+- (void)webView:(WKWebView *)webView
+    didStartProvisionalNavigation:(WKNavigation *)navigation {
+  if (webView.URL) {
+    [self addUrl:webView.URL.absoluteString];
+  }
+  //    NSLog(@"🔹 Started loading: %@", webView.URL.absoluteString);
```

Comment:
remove the unused comments, where there just to keep track over the methods that might become useful